### PR TITLE
build: run Golang module sync on dependabot PR

### PR DESCRIPTION
**Pull request by s3rj1k**
Golang module sync on dependabot PR, should force-sync Golang module data from root to API.

This is roughly based on https://github.com/dependabot/dependabot-actions-workflow/blob/main/.github/workflows/update_dependabot_bundler_pr.yml

### CODE REVIEW

File: .github/workflows/update_dependabot_bundler_pr.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# This workflow is based on https://github.com/dependabot/dependabot-actions-workflow
```
Comment (travisn) [inline .github/workflows/update_dependabot_bundler_pr.yml:1]: Are there any changes to this workflow from the examples? I don't see anything related to the modules check.

File: .github/workflows/build_dependabot_bundler_pr.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Golang
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.19
+
+      - name: run mod check
```
Comment (travisn) [inline .github/workflows/build_dependabot_bundler_pr.yml:20]: Are lines 15-21 the only lines added from the original PR? Or are there other changes?

File: .github/workflows/build_dependabot_bundler_pr.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+      - name: Configure git
+        run: |
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+
+      # NOTE: Prefixing/appending commit messages with `[dependabot skip]` allows
+      # dependabot to rebase/update the pull request, force-pushing over any changes
+
+      - name: Commit and copy license changes
```
Comment (travisn) [inline .github/workflows/build_dependabot_bundler_pr.yml:36]: Rook doesn't have `.licenses` so maybe we remove this? Or I guess it's harmless.

File: .github/workflows/build_dependabot_bundler_pr.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
+# This workflow is based on https://github.com/dependabot/dependabot-actions-workflow
+
+name: Build Dependabot Bundler PR
+on:
+  push:
+    branches:
+      - "dependabot/bundler**"
```
Comment (travisn) [inline .github/workflows/build_dependabot_bundler_pr.yml:7]: There is a bundler branch created? In rook's dependabot PRs I don't see a bundler branch in the name.

Comment (s3rj1k) [reply .github/workflows/build_dependabot_bundler_pr.yml:7]: I think that this is some magic of dependabot, according to their article

File: .github/workflows/build_dependabot_bundler_pr.yml (hunk 2 above)
Comment (s3rj1k) [reply .github/workflows/build_dependabot_bundler_pr.yml:20]: Also fixed the `- name: Commit and copy changes` item

**Comment by s3rj1k** [comment]
I am not sure how we would even test that? Maybe merge to some other that main branch and wait till dependabot PR?

**Comment by travisn** [comment]
Could you also provide a brief description paragraph in the commit message as seen in the [dev guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)? It helps in the long term to see more details in the commits for why changes were needed.

> I am not sure how we would even test that? Maybe merge to some other that main branch and wait till dependabot PR?

The dependabot only works off the master branch, so I'm not sure how to test it either without just merging it.

**Comment by s3rj1k** [comment]
> Could you also provide a brief description paragraph in the commit message as seen in the [dev guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)? It helps in the long term to see more details in the commits for why changes were needed.

Done.

**Comment by travisn** [comment]
@galexrt @BlaineEXE @subhamkrai Any suggestions on how to validate this integration with dependabot before we merge?

File: .github/workflows/build_dependabot_bundler_pr.yml (hunk 4 above)
Comment (travisn) [reply .github/workflows/build_dependabot_bundler_pr.yml:7]: Interestingly, the dependabot PRs in the rook.github.io repo such as [this](https://github.com/rook/rook.github.io/pull/148) do have the `bundler` branch.

**Comment by galexrt** [comment]
@travisn We could try merging it to a separate branch though the "easiest" for making sure it really works would be to create a standalone copy of the Rook repo (with some older dependencies), merge the changes in it, and see if the CI workflows are doing as expected.

**Comment by travisn** [comment]
> @travisn We could try merging it to a separate branch though the "easiest" for making sure it really works would be to create a standalone copy of the Rook repo (with some older dependencies), merge the changes in it, and see if the CI workflows are doing as expected.

@galexrt Interesting idea to open a new repo and test it out there. But I think the actions are scoped enough that it's low risk enough to go ahead and merge, then see if a follow-up PR is needed. Any objection, or would you like to test your suggestion?

**Comment by galexrt** [comment]
@travisn Yeah, it is low-risk enough to merge and "fix up" later if needed.

**Comment by travisn** [comment]
> @travisn Yeah, it is low-risk enough to merge and "fix up" later if needed.

Ok, let's go for it

**Pull request merged**